README.md

```diff
@@ -18,6 +18,62 @@ ArmoniK.Core provides services for submitting computational tasks, keeping track
 
+## Development process
+
+This repository follows trunk based development so all features will be integrated into main, in multiple small steps.
+Breaking changes will be limited as much as possible or gated behind feature flags.
+If breaking changes should happen, they will be documented in the releases.
+Therefore, features in development or testing that cannot fit within a branch and need to be integrated into main will be marked as preview so that users know that these features are still in development and subject to changes.
+
+**General availability:** The APIs, plugins and features that are generally available are highly curated and for which support is available.
+**Preview:** The APIs, plugins and features that are available on main but still in development. Their API should be stable enough for advanced users who want to test them before GA. Breaking changes should be minimal and will be notified in the release notes. Any feedback or issue encountered with these features are welcome !
+**Beta (Internal preview):** The APIs, plugins and features in active development. They are not stable enough and breaking changes may occur without any notice. Breaking changes will not always be documented. They can be used by the community for testing purposes. Before to be moved to the "Preview" stage, issues might be closed with the minimal messages "Under Active Development".
+**Deprecated:** The deprecated APIs, plugins and features. Features are flagged for future removal. Only production critical bug fixes will be considered.
+
+| Plugins  | Type           | Description                                   | Status  |
+|----------|----------------|-----------------------------------------------|---------|
+| RabbitMQ | Queue          | RabbitMQ client using AMQP protocol 0.9.2     | GA      |
+| AMQP     | Queue          | AMQP client using AMQP protocol 1.0.0         | GA      |
+| PubSub   | Queue          | Google PubSub client                          | GA      |
+| SQS      | Queue          | AWS SQS client                                | Preview |
+| Redis    | Object Storage | Redis client used to store binary data        | GA      |
+| Local    | Object Storage | File system used to store binary data         | GA      |
+| S3       | Object Storage | AWS S3 client used to store binary data       | GA      |
+| MongoDB  | Database       | MongoDB client to store ArmoniK internal data | GA      |
+
+| APIs                          | Description                                                                         | Status     |
+|-------------------------------|-------------------------------------------------------------------------------------|------------|
+| Agent.CreateTask              | Streamed RPC to submit payloads and tasks from the worker                           | Deprecated |
+| Agent.Get*Data                | RPC to get special Results                                                          | Beta       |
```

Comment:
Are they in beta even ? They are not implemented at all iirc

Reply:
Not all so they'll be in WIP^^